reader_test.go

```diff
@@ -10,93 +10,74 @@ import (
 	"testing"
 	"time"
 
-	. "gopkg.in/check.v1"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestMaxMindDbReader(t *testing.T) { TestingT(t) }
-
-type MySuite struct{}
-
-var _ = Suite(&MySuite{})
-
-func (s *MySuite) TestReader(c *C) {
+func TestReader(t *testing.T) {
 	for _, recordSize := range []uint{24, 28, 32} {
 		for _, ipVersion := range []uint{4, 6} {
 			fileName := fmt.Sprintf("test-data/test-data/MaxMind-DB-test-ipv%d-%d.mmdb", ipVersion, recordSize)
 			reader, err := Open(fileName)
-			if err != nil {
-				c.Logf("unexpected error while opening database: %v", err)
-				c.Fail()
-			}
-
-			checkMetadata(c, reader, ipVersion, recordSize)
+			assert.Nil(t, err, "unexpected error while opening database: %v", err)
+			checkMetadata(t, reader, ipVersion, recordSize)
 
 			if ipVersion == 4 {
-				checkIpv4(c, reader)
+				checkIpv4(t, reader)
 			} else {
-				checkIpv6(c, reader)
+				checkIpv6(t, reader)
 			}
 		}
 	}
 }
 
-func (s *MySuite) TestReaderBytes(c *C) {
+func TestReaderBytes(t *testing.T) {
 	for _, recordSize := range []uint{24, 28, 32} {
 		for _, ipVersion := range []uint{4, 6} {
 			fileName := fmt.Sprintf("test-data/test-data/MaxMind-DB-test-ipv%d-%d.mmdb", ipVersion, recordSize)
 			bytes, _ := ioutil.ReadFile(fileName)
 			reader, err := FromBytes(bytes)
-			if err != nil {
-				c.Logf("unexpected error while opening bytes: %v", err)
-				c.Fail()
-			}
+			assert.Nil(t, err, "unexpected error while opening bytes: %v", err)
 
-			checkMetadata(c, reader, ipVersion, recordSize)
+			checkMetadata(t, reader, ipVersion, recordSize)
 
 			if ipVersion == 4 {
-				checkIpv4(c, reader)
+				checkIpv4(t, reader)
 			} else {
-				checkIpv6(c, reader)
+				checkIpv6(t, reader)
 			}
 		}
 	}
 }
 
-func (s *MySuite) TestDecodingToInterface(c *C) {
+func TestDecodingToInterface(t *testing.T) {
 	reader, err := Open("test-data/test-data/MaxMind-DB-test-decoder.mmdb")
-	if err != nil {
-		c.Logf("unexpected error while opening database: %v", err)
-		c.Fail()
-	}
+	assert.Nil(t, err, "unexpected error while opening database: %v", err)
 
 	var recordInterface interface{}
 	err = reader.Lookup(net.ParseIP("::1.1.1.0"), &recordInterface)
-	if err != nil {
-		c.Logf("unexpected error while doing lookup: %v", err)
-		c.Fail()
-	}
-	record := recordInterface.(map[string]interface{})
+	assert.Nil(t, err, "unexpected error while doing lookup: %v", err)
```

Comment:
These `err` checks are probably better off as `require.Nil` instead of ``assert.Nil` as continuing will just produce confusing failures, but I can go through and change that.